superset-frontend/packages/superset-ui-chart-controls/src/types.ts

```diff
@@ -66,6 +67,7 @@ export interface Dataset {
   id: number;
   type: DatasourceType;
   columns: ColumnMeta[];
+  column_types?: GenericDataType[];
```

Comment:
the `GenericDataType` has been in ColumnMeta.

Reply:
You're right. Removed.

Reply:
I had to revert the change because columns and column_types don't match. I investigated but wasn't able to find the root cause. I believe merging these will require its own PR and tests so I'll leave it out of this PR.

Reply:
Running E2E tests, I found a case where the `columns` were all strings but `column_types` contained 0, 1, 2 (numeric, string, temporal)